src/basic/string-util.c

```diff
@@ -521,6 +521,18 @@ char* strshorten(char *s, size_t l) {
         return s;
 }
 
+int strgrowpad0(char **s, size_t l) {
+        assert(s);
+
+        char *q = realloc(*s, l);
+        if (!q)
+                return -ENOMEM;
+        *s = q;
+
+        memzero(*s + strlen(*s), l - strlen(*s));
```

Comment:
i figure gcc will be smart enough to execute strlen() only once, but I personally would have used a local variable to store the strlen() result between both uses...